src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -344,9 +344,16 @@ private function getTagVersions(array $tagsByKey, array &$invalidatedTags = arra
         $tags = array();
         foreach ($tagVersions as $tag => $version) {
             $tags[$tag.static::TAGS_PREFIX] = $tag;
-            if ($fetchTagVersions || !isset($this->knownTagVersions[$tag])) {
+
+            if ($fetchTagVersions) {
+                continue;
+            }
+
+            if (!isset($this->knownTagVersions[$tag])) {
+                $fetchTagVersions = true;
```

Comment:
That's the correct fix thanks. I'd just suggest to keep the previous single "if" and add this line in it.